apps/client/src/components/SearchDialog/index.tsx

```diff
@@ -0,0 +1,170 @@
+import { useState, useEffect } from 'react';
+
+import { SearchIcon } from 'client/assets';
+
+import { Button, Input } from 'shared/components';
+import { Dialog, DialogContent, DialogTrigger, DialogTitle, DialogClose } from 'shared/components';
+import { cn } from 'shared/lib/utils';
+
+interface Props {
+  keyword: string;
+  setKeyword: React.Dispatch<React.SetStateAction<string>>;
+}
+
+interface SchoolType {
+  SCHUL_NM: string;
+  ORG_RDNMA: string;
+  LCTN_SC_NM: string;
+}
+
+const searchElements = (schools: SchoolType[], onSelectSchool: (school: SchoolType) => void) => {
+  if (schools.length === 0) return;
+
+  return (
+    <div
+      className={cn(
+        'absolute',
+        'top-full',
+        'left-0',
+        'right-0',
+        'mt-1',
+        'flex',
+        'flex-col',
+        'items-start',
+        'rounded-md',
+        'border',
+        'border-solid',
+        'border-zinc-200',
+        'bg-white',
+        'shadow-md',
+        'overflow-y-auto',
+        'max-h-32',
+      )}
+    >
+      {schools.map((school, index) => (
+        <div
+          key={index}
+          className={cn(
+            'flex',
+            'w-full',
+            'py-2',
+            'px-2.5',
+            'items-center',
+            'cursor-pointer',
+            'hover:bg-zinc-100',
+          )}
+          onClick={() => onSelectSchool(school)}
+        >
+          <p className={cn('text-zinc-950', 'text-body1', 'font-normal')}>{school.SCHUL_NM}</p>
+        </div>
+      ))}
+    </div>
+  );
+};
+
+const SearchDialog = ({ keyword, setKeyword }: Props) => {
+  const [schools, setSchools] = useState<SchoolType[]>([]);
+  const [debounceTimeout, setDebounceTimeout] = useState<NodeJS.Timeout | null>(null);
+  const [isSelecting, setIsSelecting] = useState<boolean>(false);
+
+  const getSchools = async () => {
+    if (keyword.trim() === '') {
+      setSchools([]);
+      return;
+    }
+
+    const response = await fetch(
+      new URL(
+        `https://open.neis.go.kr/hub/schoolInfo?KEY=${process.env.NEXT_PUBLIC_NEIS_API_KEY}&Type=json&SCHUL_NM=${keyword}&SCHUL_KND_SC_NM='중학교'`,
+      ),
+      {
+        method: 'GET',
+      },
+    );
+
+    const data = await response.json();
+
+    const rows = data?.schoolInfo?.[1]?.row || [];
+
+    setSchools(rows);
+  };
+
+  useEffect(() => {
+    if (isSelecting) {
+      setIsSelecting(false);
+      return;
+    }
+
+    if (debounceTimeout) {
+      clearTimeout(debounceTimeout);
+    }
+
+    const timeout = setTimeout(() => {
+      // eslint-disable-next-line no-void
+      void getSchools();
+    }, 400);
+
+    setDebounceTimeout(timeout);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [keyword]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setKeyword(e.target.value);
+  };
+
+  const handleSelectSchool = (school: SchoolType) => {
```

Comment:
위 함수랑 동일한 형식을 가지려면 이렇게 변경하는게 좋을 것 같네요!

```suggestion
  const handleSchoolSelect = (school: SchoolType) => {
```